src/lib/operators/jit_operator/jit_operations.hpp

```diff
@@ -226,13 +289,30 @@ __attribute__((noinline)) void jit_aggregate_compute(const T& op_func, const Jit
 
   const auto catching_func = InvalidTypeCatcher<decltype(store_result_wrapper), void>(store_result_wrapper);
 
+  // this is only the case for lhs = int and rhs = float, which become long / double when aggregate sum/avg is used
```

Comment:
Why not the other way round?

Reply:
This is only used for aggregates with `sum` and `avg` on `int` and `float` columns as aggregate columns always use 64 bit types. See comment below.